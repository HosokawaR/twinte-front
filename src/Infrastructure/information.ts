import marked from 'marked'
import { OutputInformationData } from '~/api/@types'

export class Information {
  private parse(infos: OutputInformationData[]): OutputInformationData[] {
    return infos.map((info) => ({ ...info, content: marked(info.content) }))
  }

  async getInfo() {
    const response = await $nuxt.$api.information.$get()
    return this.parse(response)
  }
}
