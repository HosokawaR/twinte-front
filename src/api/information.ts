import axios, { AxiosResponse } from 'axios'
import marked from 'marked'

import { BASE_URL } from '../common/config'

export type Information = {
  id: string
  title: string
  content: string
  date: string
}[]

/**
 * 運営からのお知らせを取得する
 *
 * [
 *   {
 *     "date": "2020-06-14 20:00:00",
 *     "id": "eab4dbd6-7e52-45a5-8c92-23b9bdde9226",
 *     "title": "お知らせ機能を追加しました！",
 *     "content": "# Twin:te（ついんて）とは？\n\n## 筑波大生による筑波大生のための時間割アプリ\n\n筑波大生に最適な設計になっており、いかに時間割を簡単に作るかに重きを置いています。\nわずか数ステップで時間割を作成・管理することができます。\n\n\n現在もOSSとして保守・アップデートが行われているため、新たな機能を提案していただくことも可能です。\n\n## Twinsから安全に時間割を自動登録\n\nTwin:teではTwinsの時間割データを用いて簡単に時間割を作成することができます。\n また昨今の危険性が指摘されている時間割アプリのように、学籍番号やパスワードといった情報をサーバー側で取得することはございません。\n（どのような動作原理なのかは [開発者ブログ](https://twinte.hatenablog.com/)をご参照ください。また全てのソースコードを[Github](https://github.com/twin-te) にて公開しています。）",
 *     "tag": "info"
 *   }
 * ]
 */
export const getInformation = async (
  limit = 50,
  offset = 0
): Promise<AxiosResponse<Information>> => {
  const url = `${BASE_URL}/information/`

  return axios.get<Information>(url, {
    params: {
      limit,
      offset,
    },
  })
}

/**
 * markdownをHTMLに変換
 */
export const parse = (infos: Information): Information => {
  return infos.map((info) => ({ ...info, content: marked(info.content) }))
}
