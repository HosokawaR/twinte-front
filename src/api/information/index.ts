/* eslint-disable */
import * as Types from '../@types'

export type Methods = {
  get: {
    query?: {
      limit?: number
      offset?: number
    }

    status: 200
    resBody: Types.OutputInformationData[]
  }
}
