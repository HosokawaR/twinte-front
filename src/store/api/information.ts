import marked from 'marked'
import { OutputInformationData } from '~/api/@types'

export class Information {
  private info: OutputInformationData[] = []

  private parse(infos: OutputInformationData[]): OutputInformationData[] {
    return infos.map((info) => ({ ...info, content: marked(info.content) }))
  }

  async getInfo() {
    const response = await $nuxt.$api.information.$get()
    this.info = this.parse(response)
    return this.info
  }
}
