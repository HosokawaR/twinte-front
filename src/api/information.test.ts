import { getInformation } from './information'

jest.setTimeout(30000)

describe(getInformation, () => {
  it('basic usage', async () => {
    const info = await getInformation()
    expect(info.status).toBe(200)
  })
})
