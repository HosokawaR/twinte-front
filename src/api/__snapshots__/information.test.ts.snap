// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic usage: info__result 1`] = `
Object {
  "content": "筑波大生に最適な設計になっており、いかに時間割を簡単に作るかに重きを置いています。
わずか数ステップで時間割を作成・管理することができます。


現在もOSSとして保守・アップデートが行われているため、新たな機能を提案していただくことも可能です。

## Twinsから安全に時間割を自動登録

Twin:teではTwinsの時間割データを用いて簡単に時間割を作成することができます。
 また昨今の危険性が指摘されている時間割アプリのように、学籍番号やパスワードといった情報をサーバー側で取得することはございません。
（どのような動作原理なのかは [開発者ブログ](https://twinte.hatenablog.com/)をご参照ください。また全てのソースコードを[Github](https://github.com/twin-te) にて公開しています。）",
  "date": "2020-06-14 20:00:00",
  "id": "eab4dbd6-7e52-45a5-8c92-23b9bdde9226",
  "tag": "info",
  "title": "お知らせ機能を追加しました！",
}
`;

exports[`parse is Endomorphism: info__result__parced 1`] = `
Object {
  "content": "<p>筑波大生に最適な設計になっており、いかに時間割を簡単に作るかに重きを置いています。
わずか数ステップで時間割を作成・管理することができます。</p>
<p>現在もOSSとして保守・アップデートが行われているため、新たな機能を提案していただくことも可能です。</p>
<h2 id=\\"twinsから安全に時間割を自動登録\\">Twinsから安全に時間割を自動登録</h2>
<p>Twin:teではTwinsの時間割データを用いて簡単に時間割を作成することができます。
 また昨今の危険性が指摘されている時間割アプリのように、学籍番号やパスワードといった情報をサーバー側で取得することはございません。
（どのような動作原理なのかは <a href=\\"https://twinte.hatenablog.com/\\">開発者ブログ</a>をご参照ください。また全てのソースコードを<a href=\\"https://github.com/twin-te\\">Github</a> にて公開しています。）</p>
",
  "date": "2020-06-14 20:00:00",
  "id": "eab4dbd6-7e52-45a5-8c92-23b9bdde9226",
  "tag": "info",
  "title": "お知らせ機能を追加しました！",
}
`;
